astropy/time/core.py

```diff
@@ -1928,19 +1928,19 @@ def now(cls):
         method is called.
 
         .. note::
-            "Now" is determined using the `~datetime.datetime.utcnow`
+            "Now" is determined using the `~datetime.datetime.now`
             function, so its accuracy and precision is determined by that
             function.  Generally that means it is set by the accuracy of
-            your system clock.
+            your system clock. Timezone is set to UTC.
 
         Returns
         -------
         nowtime : :class:`~astropy.time.Time`
             A new `Time` object (or a subclass of `Time` if this is called from
             such a subclass) at the current time.
         """
-        # call `utcnow` immediately to be sure it's ASAP
-        dtnow = datetime.utcnow()
+        # call `now` immediately to be sure it's ASAP
+        dtnow = datetime.now(tz=timezone.utc)
```

Comment:
I think a new (optional) argument should be added to ``now`` that allows the user to specify the timezone.
The default value should be gotten from ``time.tzname``, corrected for daylight savings with ``'time.localtime().tm_isdst`` (see https://stackoverflow.com/questions/1111056/get-time-zone-information-of-the-system-in-python and https://bugs.python.org/issue22798).

Reply:
I think `time.localtime().tm_gmtoff` will be necessary to get the offset from utc

Reply:
daylight savings corrections might use `tzinfo.dst` instead of `tm_isdst`. I haven't investigated details

Reply:
I like the idea of an optional argument, but it would be a big API change to make `Time.now()` return non-UTC times, so I think the default has to be `timezone.utc`. Obviously, this could also be done as follow-up, since here the idea is just to avoid the `utcnow()` method. Indeed, given the need to get this in for #14784, perhaps it is best to make minimal changes here.